src/allmydata/test/test_time_format.py

```diff
@@ -0,0 +1,169 @@
+        # time_format.iso_utc_time_to_localseconds() breaks if the timezone is
+        # Europe/London.  (As soon as this unit test is done then I'll change
+        # that implementation to something that works even in this case...)
+
+        if not self.have_working_tzset():
+            raise unittest.SkipTest("This test can't be run on a platform without time.tzset().")
+
+        self.setTimezone("Europe/London")
+        return self._help_test_epoch()
+
+    def _help_test_epoch(self):
+        origtzname = time.tzname
+        s = time_format.iso_utc_time_to_seconds("1970-01-01T00:00:01")
+        self.failUnlessEqual(s, 1.0)
+        s = time_format.iso_utc_time_to_seconds("1970-01-01_00:00:01")
+        self.failUnlessEqual(s, 1.0)
+        s = time_format.iso_utc_time_to_seconds("1970-01-01 00:00:01")
+        self.failUnlessEqual(s, 1.0)
+
+        self.failUnlessEqual(time_format.iso_utc(1.0), "1970-01-01_00:00:01")
+        self.failUnlessEqual(time_format.iso_utc(1.0, sep=" "),
+                             "1970-01-01 00:00:01")
+
+        now = time.time()
+        isostr = time_format.iso_utc(now)
+        timestamp = time_format.iso_utc_time_to_seconds(isostr)
+        self.failUnlessEqual(int(timestamp), int(now))
+
+        def my_time():
+            return 1.0
+        self.failUnlessEqual(time_format.iso_utc(t=my_time),
+                             "1970-01-01_00:00:01")
+        e = self.failUnlessRaises(ValueError,
+                                  time_format.iso_utc_time_to_seconds,
+                                  "invalid timestring")
+        self.failUnless("not a complete ISO8601 timestamp" in str(e))
+        s = time_format.iso_utc_time_to_seconds("1970-01-01_00:00:01.500")
+        self.failUnlessEqual(s, 1.5)
+
+        # Look for daylight-savings-related errors.
+        thatmomentinmarch = time_format.iso_utc_time_to_seconds("2009-03-20 21:49:02.226536")
+        self.failUnlessEqual(thatmomentinmarch, 1237585742.226536)
+        self.failUnlessEqual(origtzname, time.tzname)
+
+    def test_iso_utc(self):
+        when = 1266760143.7841301
+        out = time_format.iso_utc_date(when)
+        self.failUnlessEqual(out, "2010-02-21")
+        out = time_format.iso_utc_date(t=lambda: when)
+        self.failUnlessEqual(out, "2010-02-21")
+        out = time_format.iso_utc(when)
+        self.failUnlessEqual(out, "2010-02-21_13:49:03.784130")
+        out = time_format.iso_utc(when, sep="-")
+        self.failUnlessEqual(out, "2010-02-21-13:49:03.784130")
+
+    def test_parse_duration(self):
+        p = time_format.parse_duration
+        DAY = 24*60*60
+        MONTH = 31*DAY
+        YEAR = 365*DAY
+        self.failUnlessEqual(p("1 day"), DAY)
+        self.failUnlessEqual(p("2 days"), 2*DAY)
+        self.failUnlessEqual(p("3 months"), 3*31*DAY)
+        self.failUnlessEqual(p("4 mo"), 4*31*DAY)
+        self.failUnlessEqual(p("5 years"), 5*365*DAY)
```

Comment:
As long as `MONTH` and `YEAR` are introduced above, perhaps using them instead of multiplying out DAY some more here would be cool.  I know these were here before but since you drew my attention to them by merging the two different `test_parse_duration` tests (thanks for that!) you get my comment. :)

Reply:
OK.